File: ehc_customization/ehc_customization/customizations/designation/designation.py
import frappe
from ehc_customization.ehc_customization.customizations.department.doc_events.utility_functions import transfer_entry

@frappe.whitelist()
def designation_details(responsibility):
	emp_des_doc=frappe.get_doc("Employee Responsibility",responsibility)
	final_list=[]
	for i in emp_des_doc.responsibility:
		final_list.append(i)
	return final_list

def insert_after(self, method=None):
    transfer_entry(self, method)

def update(doc, method=None):
    transfer_entry(doc, method)

def delete(doc, method=None):
    transfer_entry(doc, method)

File: ehc_customization/ehc_customization/customizations/designation/designation.js

frappe.ui.form.on('Designation', {
    employee_responsibility: function (frm) {
        update_table(frm)
    },

});


function update_table(frm, field) {
    if (frm.doc.employee_responsibility) {
        frappe.call({
            method: "ehc_customization.ehc_customization.customizations.designation.designation.designation_details",
            args: {
                responsibility: frm.doc.employee_responsibility,
            },
            callback: function (r) {
                frm.doc.transfer_details = [];
                r.message.forEach(function (message) {

                    var new_row = frm.add_child('transfer_details');

                    new_row.property = message["property"];
                    new_row.type = message["type"];
                });
                refresh_field("transfer_details");
            },

        });
    }
    else {
        frm.doc.transfer_details = []
        refresh_field("transfer_details");

    }
}